go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -0,0 +1,664 @@
+		}
+		ops = append(ops, addrSetOps...)
+	}
+	for _, rule := range desiredANPState.egressRules {
+		asIndex := GetANPPeerAddrSetDbIDs(desiredANPState.name, rule.gressPrefix, fmt.Sprintf("%d", rule.gressIndex), c.controllerName, isBanp)
+		_, addrSetOps, err := c.addressSetFactory.NewAddressSetOps(asIndex, util.StringsToIPs(rule.podIPs.UnsortedList()))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create address-sets for ANP %s's"+
+				" egress rule %s/%s/%d: %v", desiredANPState.name, rule.name, rule.gressPrefix, rule.priority, err)
+		}
+		ops = append(ops, addrSetOps...)
+	}
+	return ops, nil
+}
+
+// constructOpsForPeerChanges takes the desired and current rules of the anp and returns the corresponding ops
+// for updating NBDB AddressSetobjects for those peers
+// This should be called if namespace/pod is being created/updated
+func (c *Controller) constructOpsForPeerChanges(desiredRules, currentRules []*gressRule, atLeastOneRuleUpdated *bool,
+	anpName string, isBanp bool) ([]ovsdb.Operation, error) {
+	var ops []ovsdb.Operation
+	for i := range desiredRules {
+		desiredRule := desiredRules[i]
+		currentRule := currentRules[i]
+		ipsToAdd := desiredRule.podIPs.Difference(currentRule.podIPs)
+		asIndex := GetANPPeerAddrSetDbIDs(anpName, desiredRule.gressPrefix, fmt.Sprintf("%d", desiredRule.gressIndex), c.controllerName, isBanp)
+		if len(ipsToAdd) > 0 {
+			as, err := c.addressSetFactory.GetAddressSet(asIndex)
+			if err != nil {
+				return nil, fmt.Errorf("cannot ensure that addressSet %+v exists: err %v", asIndex.GetExternalIDs(), err)
+			}
+			klog.Infof("Adding podIPs %+v to address-set %s for ANP %s", ipsToAdd, as.GetName(), anpName)
+			addrOps, err := as.AddIPsReturnOps(util.StringsToIPs(ipsToAdd.UnsortedList()))
+			if err != nil {
+				return nil, fmt.Errorf("failed to construct address-set %s's IP add ops for anp %s's rule"+
+					" %s/%s/%d: %v", as.GetName(), anpName, desiredRule.name,
+					desiredRule.gressPrefix, desiredRule.priority, err)
+			}
+			ops = append(ops, addrOps...)
+		}
+		ipsToRemove := currentRule.podIPs.Difference(desiredRule.podIPs)
+		if len(ipsToRemove) > 0 {
+			as, err := c.addressSetFactory.GetAddressSet(asIndex)
+			if err != nil {
+				return nil, fmt.Errorf("cannot ensure that addressSet %+v exists: err %v", asIndex.GetExternalIDs(), err)
+			}
+			klog.Infof("Deleting podIPs %+v from address-set %s for ANP %s", ipsToRemove, as.GetName(), anpName)
+			addrOps, err := as.DeleteIPsReturnOps(util.StringsToIPs(ipsToRemove.UnsortedList()))
+			if err != nil {
+				return nil, fmt.Errorf("failed to construct address-set %s's IP delete ops for anp %s's rule"+
+					" %s/%s/%d: %v", as.GetName(), anpName, desiredRule.name,
+					desiredRule.gressPrefix, desiredRule.priority, err)
+			}
+			ops = append(ops, addrOps...)
+		}
+		if !*atLeastOneRuleUpdated && currentRule.action != desiredRule.action || !reflect.DeepEqual(desiredRule.ports, currentRule.ports) {
```

Comment:
I guess we are ok doing an ordered comparison of ports here rather than unordered?

Reply:
Are they? I thought only rule index is part of the ID.

Reply:
Oh wait, it is since the rule index is being coded `%d.%d`, kinda confusing that it is not a separate id or something, a comment in the object ids would be welcome for this

Reply:
yeah i do like keeping the address-set bits seperate from the ACLs logic..but convertANPRuleToACL is a bit higher up lmc if i can move this to a more prominient place?

I probably messed up the namings when i moved code, so 
PeerChanges => specific add/delete IPs from existing address-sets (namespace/pod updates)
SubjectChanges => same thing as above but for LSPs.

RuleChanges => i think today this means entirely changing the rules entirely like deleting and appending where we do the full recompute..
I need another one for ActionOrPortChanges? :)

Reply:
@npinaeva and I spoke, adapting the code according to her suggestion so that ACLs are all done in one place and AS-es are all done in one place.

Reply:
this comment is fixed, just rerunning the tests to make sure we are good before I resolve this.

Reply:
one test is failing.. let me investigate:
```
        --- FAIL: TestNetworkPolicyV2Conformance/AdminNetworkPolicyIngressSCTP/Should_support_a_'deny-ingress'_policy_for_SCTP_protocol_at_the_specified_port (0.29s)

```
i must have done some wrong move.

Reply:
```
=== RUN   TestNetworkPolicyV2Conformance/AdminNetworkPolicyIngressSCTP/Should_support_a_'deny-ingress'_policy_for_SCTP_protocol_at_the_specified_port
    admin-network-policy-core-ingress-sctp-rules.go:140: FAILED Command was [/agnhost connect --timeout=3s --protocol=sctp 10.244.1.4:9003]
    admin-network-policy-core-ingress-sctp-rules.go:140: Expected connection to be dropped from network-policy-conformance-slytherin/draco-malfoy-0 to 10.244.1.4, but instead it returned a different status: OTHER: operation already in progress
    admin-network-policy-core-ingress-sctp-rules.go:142: 
        	Error Trace:	/home/surya/go/src/github.com/ovn-org/ovn-kubernetes/test/conformance/admin-network-policy-core-ingress-sctp-rules.go:142
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestNetworkPolicyV2Conformance/AdminNetworkPolicyIngressSCTP/Should_support_a_'deny-ingress'_policy_for_SCTP_protocol_at_the_specified_port
```
huh.. seems unrelated... rerunning the test

Reply:
passes conformance, resolving comment..
i need to see what this means if we see this again.